.github/cloudbuild/android.yaml

```diff
@@ -0,0 +1,111 @@
+steps:
+# We need this step to correctly identify dockerfile tag
+- id: fetch-repo-history
+  name: gcr.io/cloud-builders/git
+  timeout: 10s
+  args:
+  - fetch
+  - --unshallow
+# Download DOCKERFILE_COMMIT tag if it exists to skip docker image generation,
+#   or download the :latest tag and use it as a cache,
+#   or skip downloading if :latest doesn't exist yet.
+# This being a separate step helps readability in Cloud Build console.
+- id: cache-old-image
+  name: gcr.io/cloud-builders/docker
+  timeout: 600s # 10 minutes
+  entrypoint: bash
+  args:
+  - -xc
+  - |
+    DOCKERFILE_COMMIT=$$(git log -n 1 --pretty=format:%H -- android/docker/mlperf_mobile/Dockerfile)
+    docker pull gcr.io/$PROJECT_ID/mlperf-mobile-android:$$DOCKERFILE_COMMIT \
+    || docker pull gcr.io/$PROJECT_ID/mlperf-mobile-android:latest \
+    || true
+# This step overrides the :latest tag of the image, so that we can use it in later steps.
+- id: build-new-image
+  name: gcr.io/cloud-builders/docker
+  timeout: 1800s # 30 minutes
+  entrypoint: bash
+  args:
+  - -xc
+  - |
+    DOCKERFILE_COMMIT=$$(git log -n 1 --pretty=format:%H -- android/docker/mlperf_mobile/Dockerfile)
+    docker build \
+      -t gcr.io/$PROJECT_ID/mlperf-mobile-android:$$DOCKERFILE_COMMIT \
+      -t gcr.io/$PROJECT_ID/mlperf-mobile-android:latest \
+      --cache-from gcr.io/$PROJECT_ID/mlperf-mobile-android:$$DOCKERFILE_COMMIT \

```

Comment:
It's minor optimization. I would reuse the cache from the previous commit (if we always use current commit hash to tag the image like stated in other comment).
The previous commit can be read with `git rev-parse --short HEAD~1`

Reply:
Let's discuss how to use cache and with which tags in one place. Namely, in your comment about DOCKERFILE_COMMIT.
Closing this for now.